Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py

```diff
@@ -9,22 +9,58 @@
 from lib.constants import *  # noqa: F403
 from lib.utils import follow_pointer_path, state
 
+
+class Transition(NamedTuple):
+    from_: int
+    to: int
+
+
 _possible_starting_areas = [
     area for area in ALL_TRANSITION_AREAS
-    # Remove impossible start areas + Don't immediately give TNT
+    # Remove start areas that instantly softlock you + start areas that give too much progression
     if area not in {
         LevelCRC.APU_ILLAPU_SHRINE,
         LevelCRC.SCORPION_TEMPLE,
+        LevelCRC.SCORPION_SPIRIT,
         LevelCRC.ST_CLAIRE_DAY,
+        LevelCRC.ST_CLAIRE_NIGHT,
+        LevelCRC.JAGUAR,
+        LevelCRC.PUSCA,
```

Comment:
Jag was not a valid starting area because in my version of the rando the program sends you from Jag 1 to the starting area. if the starting area is Jag, that means Jag 1 will send you to Jag 2, and Jag 2 sends you to Pusca.
Now that the code is changed to give the CONFIGS option to SKIP_JAGUAR I still believe you shouldn't have Jag as starting area. If SKIP_JAGUAR == True you obviously don't want to do the Jaguar fight, and if SKIP_JAGUAR == False you get the same problem as earlier.

I 100% had a reason to include the spirit fights in ALL_TRANSITION_AREAS, and I don't want to remove them from there. I also see no reason to exclude them from possible starting areas, as it's hilarious to press New Game and then instantly turn into a Penguin (plus it won't cause any issues so why consider this a problem?). The scorpion spirit fight is the exception, because that will instantly softlock you once it sends you to Scorpion Temple (Harry).

Spawning into a spirit fight by forcing to skip the Harry version of the level sounds completely unnecessary. What do you even gain by doing that? A 3 second timesave once in the run?

Having double the chance to start in a spirit temple also sounds completely irrelevant. Who would care about that? Doubling an extremely small chance would still result in an extremely small chance!

Grouping the "instant softlock" vs "too much progression" sounds like a good idea. That change is added to upcoming commit.

Reply:
Jag situation is already solved, Spirit/Temple situation will be fixed in upcoming commit.